server/planning/io/ingest_rule_handler.py

```diff
@@ -75,9 +75,19 @@ def apply_rule(self, rule: Dict[str, Any], ingest_item: Dict[str, Any], routing_
         if updates is not None:
             ingest_item.update(updates)
 
-        if attributes.get("autopost", False):
+        if attributes.get("autopost", False) and (updates is None or not self._is_original_posted(ingest_item)):
+            # Only autopost if:
+            # * The original has not been posted yet
+            # * Or there are no updates applied (from assigning Calendar/Agenda to the item)
+            #   because updating the Calendar/Agenda will automatically re-post the item for us
             self.process_autopost(ingest_item)
 
+    def _is_original_posted(self, ingest_item: Dict[str, Any]):
+        service = get_resource_service("events" if ingest_item[ITEM_TYPE] == CONTENT_TYPE.EVENT else "planning")
+        original = service.fine_one(req=None, _id=ingest_item.get(config.ID_FIELD))
```

Comment:
I guess the function name is incorrect here. It may be `find_one`.

Reply:
Whoops, thanks